### PR TITLE
Add hyperopt

### PR DESCRIPTION
**Pull request by krajasethupathy**

### CODE REVIEW

**Comment by krajasethupathy** [comment]
@JasonTam PR for adding hyperopt stuff

File: hpo/hyperopt_hpo.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+
+if __name__ == '__main__':
+    """
+    Runs a series of test using Hyperopt to determine which parameters are most important
+    All tests will be logged using Sacred - if the sacred database is set up correctly, otherwise it will simply run
+    To run pass in the number of hyperopt runs, the mongo db address and name
+
+    Generic:
+<<<<<<< HEAD
```
Comment (JasonTam) [inline hpo/hyperopt_hpo.py]: gotta fix these merge conflicts

File: hpo/hyperopt_hpo.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    For 5 tests to local mongo instance with collection named sacred:
+        python experiments/hyperopt_experiment.py 5 127.0.0.1:27017 sacred
+>>>>>>> 04b60841069e1011e0a6c0d323f7e9db887be9c3
+
+    """
+
+    parser = argparse.ArgumentParser(description="Titanic Hyperopt Tests logging to Sacred")
+    parser.add_argument("num_runs", type=int, help="Number of Hyperopt Runs")
+    parser.add_argument("mongo_db_address", type=str, help="Address of the Mongo DB")
```
Comment (JasonTam) [inline hpo/hyperopt_hpo.py]: add default params (like the ones suggested in README) for simplicity
(same goes for all the other args)

File: hpo/hyperopt_hpo_configs.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    # Preprocess Ingredient: preprocess
+    'preprocess': {
+        "features": hp.choice("features", [['Fare', 'SibSp'], ['Fare', 'SibSp', 'Parch']]),
+    },
+
+    # Experiment: titanic
+    'titanic': {
+        "fit_intercept": hp.choice('fit_intercept', [True, False]),
+        "penalty": hp.choice('penalty', ["l1", "l2"]),
+        "C": hp.uniform('C', .1, 1000)
```
Comment (JasonTam) [inline hpo/hyperopt_hpo_configs.py]: really should be loguniform or lognormal

**Comment by krajasethupathy** [comment]
@JasonTam made updates based on your comments

**Comment by krajasethupathy** [comment]
@JasonTam - added commit for metrics tracking. SGD's n_iter deprecated for max_iter, just fyi

**Pull request merged**